First HomeWork Васильев Максим

zservirus opened the pull request.

zservirus committed:
First HomeWork

Liberate520 commented on src/ru/gb/family/FamilyTree.java:
```diff
@@ -0,0 +1,53 @@
+        this.familyTree.add(newHuman);
+    }
+
+    public void findHuman (Human searchHuman){
```
Старайтесь возвращать результат работы методов, а не выводить в консоль

Liberate520 commented on src/ru/gb/vending/Main.java:
```diff
@@ -0,0 +1,23 @@
+package ru.gb.vending;
```
Убрать. Оставьте в этом проекте только работу с семейным деревом. Остальное в отдельные проекты

zservirus committed:
ДЗ №2
Добавили заись и чтение из файла.
Устранили ошибки 1 ДЗ..

zservirus committed:
ДЗ №3
переделал раздел генерации тестового дерева и загрузки из файла
Добавил к структуре хранения дерева новые поля. Реализовал сортировки, и поиски.

zservirus committed:
ДЗ №4
Переделали Семейное дерево под разные классы(Люди,кошки и тд..) путем обобщения

Liberate520 commented on src/ru/gb/family/ItemFamilyTrees/ItemFamilyTree.java:
```diff
@@ -0,0 +1,209 @@
+package ru.gb.family.ItemFamilyTrees;
```
перенести в пакет дерева

Liberate520 commented on src/ru/gb/family/ItemFamilyTrees/ItemFamilyTree.java:
```diff
@@ -0,0 +1,209 @@
+    private String name;
+    private Gender gender;
+    private LocalDate birthday,dateOfDeath;
+    private ItemFamilyTree<T> father;
+    private ItemFamilyTree<T> mother;
```
просто Т father

Liberate520 commented on src/ru/gb/family/ItemFamilyTrees/ItemFamilyTree.java:
```diff
@@ -0,0 +1,209 @@
+    private LocalDate birthday,dateOfDeath;
+    private ItemFamilyTree<T> father;
+    private ItemFamilyTree<T> mother;
+    private List<ItemFamilyTree<T>> children;
```
тоже самое. надо вернуть список объектов Т, а не ItemFamilyTree

Liberate520 commented on src/ru/gb/family/service/Service.java:
```diff
@@ -0,0 +1,43 @@
+
+import java.io.Serializable;
+
+public class Service<E extends FamilyTree> implements Serializable {
```
Идея класс Сервис неверна

zservirus committed:
ДЗ №4 - работа над ошибками
надо вернуть список объектов Т, а не ItemFamilyTree - поправил.

а вот "Идея класс Сервис неверна". ход мыслей встал..

zservirus committed:
Переделал идею Service

zservirus committed:
ДЗ№5

Liberate520 commented on src/ru/gb/family/model/familyTree/ItemFamilyTrees/enums/DegreeOfKinship.java:
```diff
@@ -0,0 +1,5 @@
+package ru.gb.family.model.familyTree.ItemFamilyTrees.enums;
```
неверное деление по пакетам. В джава пакет это элемент инкапсуляции, поэтому неверна идея, что enum будем класть в один пакет, интерфейсы в другой и классы в третий. Делим по пакетам по общей решаемой задаче

Liberate520 commented on src/ru/gb/family/model/familyTree/ItemFamilyTrees/genTest/GeneratItemFamilyTree.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.List;
+
+public class GeneratItemFamilyTree<T extends ItemFamilyTree<T>> {
+    public GeneratItemFamilyTree() {
+        List<ItemFamilyTree<Human>> newItemFamilyTree = new ArrayList<>();
+    }
```
В чем смысл? Создали пустой список и что с этим списком? Он никак не используется

Liberate520 commented on src/ru/gb/family/model/familyTree/ItemFamilyTrees/genTest/GeneratItemFamilyTree.java:
```diff
@@ -0,0 +1,53 @@
+    public FamilyTree gen() {
+
+
+        Service testService = new Service<FamilyTree, T>();
```
Дерево не должно создавать Сервис. Сервис использует дерево, а не наоборот

Liberate520 replied:
Ну или не дерево, а генератор людей какой то

Liberate520 commented on src/ru/gb/family/model/service/Service.java:
```diff
@@ -0,0 +1,113 @@
+import java.util.List;
+
+
+public class Service<E extends FamilyTree, T extends ItemFamilyTree<T>> implements Serializable {
```
Не надо делать обобщения для этого класса, для презентера и для View. Ориентируемся на работу с людьми

Liberate520 commented on src/ru/gb/family/model/service/Service.java:
```diff
@@ -0,0 +1,113 @@
+public class Service<E extends FamilyTree, T extends ItemFamilyTree<T>> implements Serializable {
+    private FamilyTree familyTree;
+    private long idItemFamilyTree;
+    private String pathDir = "src/ru/gb/family/model/data/";
```
Не характеризует работу Сервис. Переменная должна храниться там, где она используется

Liberate520 commented on src/ru/gb/family/model/service/Service.java:
```diff
@@ -0,0 +1,113 @@
+
+
+public class Service<E extends FamilyTree, T extends ItemFamilyTree<T>> implements Serializable {
+    private FamilyTree familyTree;
```
Тип дерева? Дерево же обобщенный класс, всегда значит надо указывать тип

Liberate520 commented on src/ru/gb/family/presenter/Presenter.java:
```diff
@@ -0,0 +1,73 @@
+
+    }
+
+    public FamilyTree getTree() {
```
Не думаю, что надо возвращать дерево во View, View ничего не должно знать о модели и не забывайте указывать тип дерева

Liberate520 commented on src/ru/gb/family/model/service/Service.java:
```diff
@@ -0,0 +1,113 @@
+
+    public void creatItemFamilyTree(String name, LocalDate birthday,LocalDate dateOfDeath, Gender gender){
+        if (dateOfDeath != null){
+            familyTree.creatItemFamilyTreeInTree(idItemFamilyTree++,name,birthday,dateOfDeath,gender);
```
Контейнеры не создают объекты хранения

zservirus committed:
ДЗ№6
Работа над ошибками и рефакторинг по принципам SOLID

Просьба посмотреть почему не происходит десериализация, при загрузке из тестового файла.. ошибок не выдает но и записи не грузит. много дней пытался разобраться.. не получилось.
